### PR TITLE
add AsDouble() 

### PR DESCRIPTION
**Pull request by MikeStall**
Add helpers to say:

`double x = value.AsDouble();`  

or AsDecimal. 

this avoids having to cast and guess whether it's decimal or double.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by jas-valgotar — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValue.cs [hunk 1]
```diff
@@ -88,6 +89,68 @@ public bool TryGetPrimitiveValue(out object val)
...
+                    return b;
+                }
+            }
+
+            throw new InvalidOperationException($"Can't coerce to double from {this.Type._type.GetKindString()})");
+        }
+
+        public double AsDouble()
+        {
+            if (this is BlankValue)
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValue.cs:114]: This is dangerous - blanks are different than 0.0, and it is ideal to mix those two concepts.

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValue.cs:114]: Caller can always get blank if they care. But a common pattern is missing is default, some should make that easy.

**Pull request merged**